File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        mp = {}
        for c in s:
            mp[c] = mp.get(c,0)+1
        for c in t:
            if c not in mp : return False
            mp[c] = mp[c] - 1
        for k in mp:
            if mp[k] != 0: return False 
        return True

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) return false;
        int board[] = new int[26];
        for(char ch : s.toCharArray()) board[ch-'a']++;
        for(char ch : t.toCharArray()) board[ch-'a']--;
        for(int i : board) if(i>0) return false;
        return true;
    }
}